stock-tracker/src/socket/eventActions.ts

```diff
@@ -15,17 +15,38 @@ import {
 import { addTopPeersAction, setErrorPeersAction } from "../features/topPeers";
 import { setSuggestionsAction, setErrorSearchAction } from "../features/search";
 import { ActionCreator } from "redux";
-import { SetChartData, SetChartError } from "features/chart/redux/actions";
+import {
+  SetChartData,
+  SetChartError,
+  ChartData
+} from "features/chart/redux/actions";
 import {
   SetCompanyOverview,
-  SetErrorCompanyOverview
+  SetErrorCompanyOverview,
+  CompanyOverviewData
 } from "features/overview/redux/actions";
-import { SetLatestNews, SetErrorNews } from "features/latestNews/redux/actions";
-import { SetResponse, SetErrorKeyStats } from "features/keyStats/redux/actions";
-import { SetSuggestions, SetSearchError } from "features/search/redux/actions";
-import { SetTopPeers, SetErrorPeers } from "features/topPeers/redux/actions";
+import {
+  SetLatestNews,
+  SetErrorNews,
+  LatestNews
+} from "features/latestNews/redux/actions";
+import {
+  SetResponse,
+  SetErrorKeyStats,
+  ResponseData
+} from "features/keyStats/redux/actions";
+import {
+  SetSuggestions,
+  SetSearchError,
+  SearchData
+} from "features/search/redux/actions";
+import {
+  SetTopPeers,
+  SetErrorPeers,
+  TopPeers
+} from "features/topPeers/redux/actions";
 
-export type EventActions = EventAction[];
+export type EventActions = EventAction<Events>[];
 
 export type Actions =
```

Comment:
You don't need this explicit type anymore. Can replace with:
```ts
export type Actions = keyof EventActionsMap;
```

I suspect it is probably the same with a few of these types here, in fact.

Reply:
Thanks, this was the missing link I was trying to figure out!